### ISSUE TITLE
[Bug] Continue to get database errors when I launch the update after fixing the config.yml

### ISSUE THREAD

**Issue by Gamerayers** [zurdi15/romm#715]
**RomM version**
Version of RomM you are using  (zurdi15/romm:latest)

**Describe the bug**
Did an update in unraid, and now can't launch it.

**To Reproduce**
I was using it, I updated it to the latest version. It stopped working. I added in the new config. That seemed to fix things a little, but now getting a different error.

I have tried to remove everything and start over, and I still get this error.

**Expected behavior**

**Screenshots**
![image](https://github.com/zurdi15/romm/assets/59520501/2907a643-6296-48c3-bba6-33e2e77b65d6)

**Desktop (please complete the following information):**
 - OS: Unraid - Community App
 - Browser Chrome

**Additional context**

**Comment by gantoine** [comment]
> Missing database credentials. Please check your configuration file

This is cause you're missing something in your environment variables, check the example docker-compose.example.yml.

**Comment by Gamerayers** [comment]
Nope, somebody needs to recreate the app.  See the current one doesn't support the separate database docker image. That would be nice, but it doesn't have it today.
So, the unraid app is broken until somebody recreates it with the database.

**Comment by gantoine** [comment]
Can you post a link to the unraid community app?

**Comment by Gamerayers** [comment]
https://forums.unraid.net/topic/149738-support-eurotimmy-romm-rom-manager-by-zurdi15/

**Comment by Gamerayers** [comment]
It appears somebody has added some additional documentation in the forum that now says you need the MariaDB separate of it. It previously was not required.

**Comment by zurdi15** [comment]
Correct, that's because previous to 3.0.0 romm supported sqlite, but since we have a more complex (and it will grow over time) relational database squema, we decided to drop sqlite support as database and romm will only support mariadb, that's why you need to setup a mariadb container now.

In the other hand, redis container is no longer required as a separated container in order to make user management available because it's integrated inside romm's container

**Comment by zurdi15** [comment]
> Can you post a link to the unraid community app?

https://unraid.net/community/apps?q=romm#r

**Comment by iamonlv426** [comment]
I'm seeing the same problem with my setup, Its new (I've not used rommM previous to 3.01 release) and giving an almost identical database error. I have a number of containers that connect and make use of databases in my MariaDB container without issue.  I've checked all my variables but can't seem to rectify the issue. 

CRITICAL: [RomM][2024-03-20 11:41:14] Missing database credentials. Please check your configuration file
CRITICAL:     [RomM][romm] Missing database credentials. Please check your configuration file
ERROR:    [init][2024-03-20 11:41:14] Something went horribly wrong with our database
INFO:     [init][2024-03-20 11:41:22] starting redis-server
9:C 20 Mar 2024 11:41:22.646 * oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
9:C 20 Mar 2024 11:41:22.646 * Redis version=7.2.4, bits=64, commit=00000000, modified=0, pid=9, just started
9:C 20 Mar 2024 11:41:22.646 * Configuration loaded
9:M 20 Mar 2024 11:41:22.646 * monotonic clock: POSIX clock_gettime
9:M 20 Mar 2024 11:41:22.647 * Running mode=standalone, port=6379.
9:M 20 Mar 2024 11:41:22.647 * Server initialized
9:M 20 Mar 2024 11:41:22.647 * Ready to accept connections tcp
CRITICAL: [RomM][2024-03-20 11:41:22] Missing database credentials. Please check your configuration file
CRITICAL:     [RomM][romm] Missing database cre

**Comment by gantoine** [comment]
@iamonlv426 what does your docker-compose look like?

**Comment by iamonlv426** [comment]
Ho @gantoine appreciate your quick response and assistance - see below - hopefully it's a silly mistake. 


romm:
    image: zurdi15/romm:latest
    container_name: romm
    security_opt:
      - no-new-privileges:true
    restart: unless-stopped
    networks:
      - proxy
    ports:
      - 8282:80
    environment:
      TZ: $TZ
      PUID: $PUID
      PGID: $PGID
      DB_HOST: $DB_HOST
      DB_PORT: $DB_PORT
      DB_DATABASE: romm
      DB_USERNAME: romm_user
      DB_PASSWORD: $ROMM_PASSWORD
      MYSQL_DATABASE_FILE: /run/secrets/romm_db_name
      MYSQL_USER_FILE: /run/secrets/romm_mysql_user
      MYSQL_PASSWORD_FILE: /run/secrets/romm_mysql_password
      MYSQL_ROOT_PASSWORD: /run/secrets/mysql_root_password
      IGDB_CLIENT_ID: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx # Generate an ID and SECRET in IGDB
      IGDB_CLIENT_SECRET: xxxxxxxxxxxxxxxxxxxxxx # https://api-docs.igdb.com/#account-creation
      ROMM_AUTH_SECRET_KEY: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx # Generate a key with `openssl rand -hex 32`
      ROMM_AUTH_USERNAME: admin
      ROMM_AUTH_PASSWORD: # default: admin
    secrets:
      - romm_db_name
      - romm_mysql_user
      - romm_mysql_password
      - mysql_root_password
    volumes:
      - $DOCKERDIR/appdata/romm/config:/config # [Optional] Path where config.yml is stored
      - $DOCKERDIR/appdata/romm/data:/data
      # - romm_resources:/romm/resources # Resources fetched from IGDB (covers, screenshots, etc.)
      # - romm_redis_data:/redis-data # Cached data for background tasks
      - $DOCKERDIR/appdata/romm/data/redis-data:/redis-data:rw # Cached data for background tasks
      - $DOCKERDIR/appdata/romm/data/resources:/resources:rw # Resources fetched from IGDB (covers, screenshots, etc.)
      - $DATADIR/roms/roms:/roms/roms:ro # Your game library
      - $DATADIR/roms/assets:/roms/assets:rw # Uploaded saves, states, etc.
      - $DOCKERDIR/logs/cloudserver/romm # [Optional] Path where logs are stored

**Comment by gantoine** [comment]
i think the env section needs `-` before each line to be valid. oh and make sure `DB_HOST`, `DB_PORT` and `DB_DATABASE` are set in your .env file (or whatever you use)

**Comment by iamonlv426** [comment]
Ok I'll check but I use the same env section for many containers. Those are set correctly in my .env file

**Comment by gantoine** [comment]
oh i see the issue, you want it like this:
```
DB_USER: romm_user
DB_PASSWD: $ROMM_PASSWORD
```

```
if not DB_USER or not DB_PASSWD:
  log.critical(
    "Missing database credentials. Please check your configuration file"
  )
  sys.exit(3)
```

**Comment by iamonlv426** [comment]
Thank you that was exactly the issue. Just stupid on my behalf - thanks again.
